File: core/SyncPlatform/src/Services/Iam/Iam.Domain/Models/BiometricProfile.cs
using Libs.Shared.Common;
using Iam.Domain.Enums;

namespace Iam.Domain.Models;

public class BiometricProfile : BaseAuditableEntity
{
    public Guid UserId { get; set; }

    public virtual User User { get; set; } = null!;

    public Gender Gender { get; set; }

    public DateOnly DateOfBirth { get; set; }

    public decimal HeightCm { get; set; }

    public decimal CurrentWeightKg { get; set; }

    public decimal TargetWeightKg { get; set; }

    public decimal? CurrentBodyFatPercentage { get; set; }

    public decimal? GoalBodyFatPercentage { get; set; }

    public decimal? MuscleMassKg { get; set; }

    public FitnessGoal FitnessGoal { get; set; }

    public ActivityLevel ActivityLevel { get; set; }

    public FitnessExperienceLevel FitnessExperienceLevel { get; set; }

    public WorkoutLocationPreference WorkoutLocationPreference { get; set; }

    public int BaseTDEE { get; set; }

    public int BMR { get; set; }

    public int? DailyProteinTargetGram { get; set; }

    public int? DailyCarbTargetGram { get; set; }

    public int? DailyFatTargetGram { get; set; }

    /// <summary>
    /// Mục tiêu calo goal-adjusted. Null = chưa có (client dùng BaseTDEE như cũ).
    /// Được Adaptive Engine ghi qua apply-targets.
    /// </summary>
    public int? DailyCalorieTarget { get; set; }

    /// <summary>
    /// true = targets do Adaptive Engine quản (Recalculate không ghi đè macro/calo
    /// bằng công thức nữa — chỉ cập nhật BMR/BaseTDEE). Reset khi user đổi goal.
    /// </summary>
    public bool TargetsManagedByEngine { get; set; }

    public DateTime? TargetsAdjustedAtUtc { get; set; }

    public List<string>? Injuries { get; set; }

    public List<string>? Medications { get; set; }
}

File: ai-ecosystem/agentic-lifestyle/src/sync_agent/domain/schemas/iam/biometric_profile.py
"""Iam.Domain.Models.BiometricProfile + BaseAuditableEntity."""

from datetime import date
from uuid import UUID

from pydantic import AliasChoices, Field, field_validator

from sync_agent.domain.schemas.common import AuditableEntityFields, DateOnly, DecimalField, StrictModel
from sync_agent.domain.schemas.enums import (
    ActivityLevel,
    FitnessExperienceLevel,
    FitnessGoal,
    Gender,
    WorkoutLocationPreference,
)
from sync_agent.domain.schemas.enums._coerce import coerce_int_enum


class BiometricProfile(AuditableEntityFields):
    """
    Maps C# BiometricProfile (excludes navigation property User).
    Inherits Id, CreatedAt, UpdatedAt, DeletedAt from BaseAuditableEntity.
    """

    user_id: UUID
    gender: Gender
    date_of_birth: DateOnly
    height_cm: DecimalField
    current_weight_kg: DecimalField
    target_weight_kg: DecimalField
    current_body_fat_percentage: DecimalField | None = None
    goal_body_fat_percentage: DecimalField | None = None
    muscle_mass_kg: DecimalField | None = None
    fitness_goal: FitnessGoal
    activity_level: ActivityLevel
    fitness_experience_level: FitnessExperienceLevel
    workout_location_preference: WorkoutLocationPreference
    base_tdee: int = Field(validation_alias=AliasChoices("baseTDEE", "baseTdee"))
    bmr: int
    daily_protein_target_gram: int | None = None
    daily_carb_target_gram: int | None = None
    daily_fat_target_gram: int | None = None
    injuries: list[str] | None = None
    medications: list[str] | None = None

    @field_validator(
        "gender",
        "fitness_goal",
        "activity_level",
        "fitness_experience_level",
        "workout_location_preference",
        mode="before",
    )
    @classmethod
    def _coerce_enums(cls, value: object, info):  # type: ignore[no-untyped-def]
        enum_map = {
            "gender": Gender,
            "fitness_goal": FitnessGoal,
            "activity_level": ActivityLevel,
            "fitness_experience_level": FitnessExperienceLevel,
            "workout_location_preference": WorkoutLocationPreference,
        }
        enum_cls = enum_map[info.field_name]
        return coerce_int_enum(enum_cls, value)
